### PR TITLE
Produce a manifest for Blazor SDK

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

**Comment by mkArtakMSFT** [comment]
Sorry @pranavkm. Just tested out some automation with this PR

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in [hunk 1]
```diff
@@ -0,0 +1,33 @@
+{
+  "version": 1,
+  "workloads": {
+    "microsoft-net-sdk-blazorwebassembly-aot": {
```
Comment (marek-safar) [inline src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:4]: this is user-facing name, it should be something like `microsoft-blazor-webassembly`

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      "description": "Blazor WebAssembly AOT workload",
+      "packs": [
+        "Microsoft.NET.Runtime.MonoAOTCompiler.Task",
+        "Microsoft.NET.Runtime.WebAssembly.Sdk",
+        "Microsoft.Netcore.App.Runtime.Aot.Cross.browser-wasm"
```
Comment (marek-safar) [inline src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:9]: This is missing Browser runtime pack and whole Blazor SDK dependency

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "version": 1,
+  "workloads": {
+    "microsoft-net-sdk-blazorwebassembly-aot": {
+      "description": "Blazor WebAssembly AOT workload",
```
Comment (marek-safar) [inline src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:5]: Are you sure you want to have a different workload pack for AOT and non-AOT configuration?

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in [hunk 4]
```diff
@@ -0,0 +1,33 @@
+{
```
Comment (marek-safar) [inline src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:1]: This file needs to go to dotnet/sdk (so the versioning should happen there)

File: eng/Version.Details.xml [hunk 5]
```diff
@@ -286,6 +286,15 @@
...
       <Sha>c07d59af4edfb22ebef9a8922cca990f6242b53b</Sha>
     </Dependency>
+    <Dependency Name="Microsoft.NET.Runtime.MonoAOTCompiler.Task" Version="6.0.0-preview.2.21116.2">
+      <Uri>https://github.com/dotnet/runtime</Uri>
+      <Sha>c07d59af4edfb22ebef9a8922cca990f6242b53b</Sha>
+    </Dependency>
+    <Dependency Name="Microsoft.NET.Runtime.WebAssembly.Sdk" Version="6.0.0-preview.2.21116.2">
+      <Uri>https://github.com/dotnet/runtime</Uri>
+      <Sha>c07d59af4edfb22ebef9a8922cca990f6242b53b</Sha>
+    </Dependency>
```
Comment (marek-safar) [inline eng/Version.Details.xml:296]: Why do you need that if you already have darc runtime dependency?

**Review by marek-safar — COMMENTED** [review]
Few comments

File: eng/Version.Details.xml (hunk 5 above)
Comment (pranavkm) [reply eng/Version.Details.xml:296]: I can check with the build team, but the file usually lists every dependency individually even if they're produced from the same build pipeline.

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in (hunk 4 above)
Comment (pranavkm) [reply src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:1]: Could you elaborate on this? I'm missing something fundamental about how this gets shipped.

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in (hunk 3 above)
Comment (pranavkm) [reply src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:5]: I haven't done any work to make BlazorWebAssembly SDK an optional pack. It's still pat of the .NET installer. The way this works is that if a Blazor project enables AOT compilation, Blazor SDK wires up the AOT targets from these packs: https://github.com/dotnet/sdk/blob/master/src/BlazorWasmSdk/Targets/Microsoft.NET.Sdk.BlazorWebAssembly.Current.targets#L559-L581. 

The workflow is a developer acquires the AOT workload (and emscripten), enables AOT in their Blazor Wasm project file (`RunAOTCompilation=true`) and publishes.

Does that align with how you think about this?

Comment (marek-safar) [reply src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:5]: > Does that align with how you think about this?

It does not. I see this as Blazor WebAssembly workload, not AOT specific configuration. The end-user scenario for me is that a developer installs .NET Sdk and then goes and installs blazor-wasm workload. The workload manifest for that will include all necessary dependencies to develop and possibly also publish blazor-wasm whether it's with AOT or without it. In other words most of the tooling will be required for non-aot or hot reload setup as well.

The scenario where you install the workload and disconnect from the internet is I believe also supported.

File: src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in (hunk 4 above)
Comment (marek-safar) [reply src/Components/WebAssembly/BlazorManifest/src/WorkloadManifest.json.in:1]: I think we probably need to discuss with @dsplaisted where this should be built into. The spec only mentions the destination https://github.com/dotnet/designs/blob/9e94ff32ea43418310c0fc77eb3703492856b364/accepted/2020/workloads/workload-manifest.md#advertising

**Review by HaoK — APPROVED** [review]

**Comment by pranavkm** [comment]
@marek-safar I'm getting this in. Let's talk about this next week and we can figure out where the right place for this needs to be.

**Comment by marek-safar** [comment]
@pranavkm it's still missing the runtime pack dependency

**Pull request merged**